lib/streamlit/elements/lib/column_types.py

```diff
@@ -156,6 +156,11 @@ class ColumnConfig(TypedDict, total=False):
         Whether edited cells in the column need to have a value. If True, an edited cell
         can only be submitted if it has a value other than None. Defaults to False.
 
+    pinned: bool or None
```

Comment:
**question (non-blocking):** Will we want to support pinning on different sides of the dataframe at some point? Right now it pins to the left side, but it's not unreasonable to assume users may want to pin to the right. If that's a possibility, we may want to consider a different data type here

Reply:
Good point. This could definitely be something to consider at some point, but I think it is unlikely that we will support pinned columns on different sides anytime soon. There isn't a user request for this yet and we would need to implement some more complicated changes in the underlying grid-library. 

I believe we could still use `bool` here for pinning to left side, and eventually add a `left` / `right` literal for `pinned` as well. I think we have a few similar cases in Streamlit where we have `bool` and `literals` (e.g. `divider` for headings). And in this case we also don't have the issue with keeping the protobuf prop types backwards compatible (since it is not using specific protobuf props).